### PR TITLE
feat: Add CRD Support for Oracle Cloud Infrastructure

### PR DESCRIPTION
**Pull request by jynolen**
## Overview

This PR is the sister PR of bank-vaults/bank-vaults#2660
It adds CRD Support for Oracle Cloud Infrastructure

## Notes for reviewer

See bank-vaults/bank-vaults#2660 for complete implementation

### CODE REVIEW

File: pkg/apis/vault/v1alpha1/vault_types.go [hunk 1]
```diff
@@ -911,6 +927,15 @@ type AWSUnsealConfig struct {
 	S3SSE                string `json:"s3SSE,omitempty"`
 }
 
+// AWSUnsealConfig holds the parameters for AWS KMS based unsealing
```
Comment (csatib02) [inline pkg/apis/vault/v1alpha1/vault_types.go:930]: Please update this comment.

Comment (jynolen) [reply pkg/apis/vault/v1alpha1/vault_types.go]: Fixed

**Comment by jynolen** [comment]
Required Changes had been performed

**Comment by csatib02** [comment]
Also please rewrite this part in the makefile:

```makefile
# Dependency versions
GOLANGCI_VERSION = 1.53.3
LICENSEI_VERSION = 0.8.0
KIND_VERSION = 0.20.0
KURUN_VERSION = 0.7.0
CODE_GENERATOR_VERSION = 0.27.1
HELM_DOCS_VERSION = 1.11.0
KUSTOMIZE_VERSION = 5.1.0
CONTROLLER_TOOLS_VERSION = 0.15.0 # <-- This line
```

Then run `make generate`.
You can commit the `controller-tools-version` change too, since it had nil pointer issues in the past.

**Comment by jynolen** [comment]
Changed done

**Review by csatib02 — APPROVED** [review]
LGTM!

**Comment by ramizpolic** [comment]
Depends on https://github.com/bank-vaults/bank-vaults/pull/2660

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**